lucet-runtime/lucet-runtime-internals/src/module/dl.rs

```diff
@@ -13,6 +13,65 @@ use std::slice;
 use std::slice::from_raw_parts;
 use std::sync::Arc;
 
+use raw_cpuid::CpuId;
+
+fn check_feature_support(module_features: &ModuleFeatures) -> Result<(), Error> {
+    let cpuid = CpuId::new();
+
+    fn missing_feature(feature: &str) -> Error {
+        Error::Unsupported(format!(
```

Comment:
This should probably be a new variant on `ModuleError`, but we're already using `Unsupported` for similar things so I'm not gonna sweat it for now